1. Черновой вариант

manualcheg opened the pull request.

manualcheg committed:
1. Черновой вариант

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/favorites/domain/useCase/DelUseCaseImpl.kt:
```diff
@@ -0,0 +1,10 @@
+import ru.practicum.android.diploma.favorites.domain.repository.FavoritesRepository
+
+class DelUseCaseImpl(private val favoritesRepository: FavoritesRepository):DelUseCase {
+    override suspend fun execute(id: Int): Boolean {
```
почему возращаешь Boolean?

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/favorites/domain/useCase/AddUseCaseImpl.kt:
```diff
@@ -0,0 +1,11 @@
+import ru.practicum.android.diploma.favorites.domain.repository.FavoritesRepository
+
+class AddUseCaseImpl(private val favoritesRepository: FavoritesRepository):AddUseCase {
+    override suspend fun execute(vacancy: Vacancy): Boolean {
```
почему возращаешь Boolean?

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/favorites/domain/useCase/IsVacancyContainsOnceUseCase.kt:
```diff
@@ -0,0 +1,5 @@
+package ru.practicum.android.diploma.favorites.domain.useCase
+
+interface IsVacancyContainsOnceUseCase {
```
а зачем нужен этот useCase?

manualcheg replied:
убран

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/vacancy/ui/viewModel/VacancyViewModel.kt:
```diff
@@ -62,15 +68,21 @@ class VacancyViewModel(
+                if (inFavorites.value == true) {
+                    delUseCase.execute(id)
+                } else {
+                    addUseCase.execute(vacancy!!)
```
лучше обернуть в let, как мне сказали "!! - это когда кричишь на компилятор, а на него не надо кричать"

manualcheg replied:
сделал

manualcheg committed:
1. Разбит на отдельные AddOrDelVacancyUseCase
2. Логика вынесена во ViewModel
3. Передается vacancyId и используется внутри VacancyViewModel
4. Убран возврат Boolean-значений при удалении/добавлении вакансии в БД
5. Vacancy сохраняется внутри VacancyViewModel для использования
6. Заменён запрос удаления из БД

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/vacancy/ui/viewModel/VacancyViewModel.kt:
```diff
@@ -60,17 +67,21 @@ class VacancyViewModel(
     }
 
-    fun addOrDelFavorites(id: Int) {
+    fun addOrDelFavorites() {
```
можно переименовать в toggleFavorites, при вызове из фрагмента будет более логично

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/vacancy/ui/viewModel/VacancyViewModel.kt:
```diff
@@ -30,28 +34,31 @@ class VacancyViewModel(
     }
 
-    fun findVacancyById(id: Int) {
+    fun findVacancyById() {
```
можно переименовать в getVacancy, вызов уже без id

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/vacancy/ui/viewModel/VacancyViewModel.kt:
```diff
@@ -30,28 +34,31 @@ class VacancyViewModel(
 
-    fun shareVacancyById(id: Int) {
-        shareVacancyByIdUseCase.execute(id)
+    fun shareVacancyById() {
```
можно переименовать в shareVacancy, вызов уже без id

manualcheg committed:
1. Изменены некоторых названия методов
2. изменен запрещенный вызов Html.fromHtml() с одним аргументом
